### ISSUE TITLE
Ontrack provisioning and configuration

### ISSUE THREAD

**Issue by dcoraboeuf** [nemerosa/ontrack#441]
Document and/or implement the way a complete and running Ontrack environment can be set up, with its configuration, etc., using the DSL and/or the REST API.

**Comment by dcoraboeuf** [comment]
Reactivating the `Shell` component, polishing its argument, especially for input/output, and packaging it as executable.

Note that Ontrack must run in order to be configured through the DSL or the REST API.

**Comment by dcoraboeuf** [comment]
See also the impact on the Ontrack Jenkins plugin with having dependencies driven now by Spring Boot, in particular the logging.

**Comment by dcoraboeuf** [comment]
The `ontrack-dsl-shell.jar` is a fully executable JAR, published in GitHub release and in the Maven Central, and can be used to setup a running instance of Ontrack:

```bash
ontrack-dsl-shell.jar --url ... --user ... --password ... --file ...
```

The `--file` argument is the path to a file containing the Ontrack DSL to execute. If not set, or set to `-`, the DSL is taken from the standard input.

> Note that the JAR is a real executable, so there is no need to use `java -jar` on Unix like systems (including MacOS).

**Comment by dcoraboeuf** [comment]
The documentation will be included in the context of #435.

**Comment by dcoraboeuf** [comment]
* Will be available in 2.24
* Will be documented in the context of #435
